pkg/model/awsmodel/network.go

```diff
@@ -564,6 +568,22 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 					EgressOnlyInternetGateway: eigw,
 				})
 			}
+
+			subnets, err := b.LinkToPrivateSubnetInZone(zone)
+			if err != nil {
+				return err
+			}
+
+			for _, subnetSpec := range b.Cluster.Spec.Subnets {
+				for _, subnet := range subnets {
+					if strings.HasPrefix(*subnet.Name, subnetSpec.Name) {
+						err := addAdditionalRoutes(subnetSpec.AdditionalRoutes, subnetSpec.Name, rt, b.Lifecycle, c)
```

Comment:
we're adding these routes to private subnets but `addAdditionalRoutes` includes `public` in the task names, perhaps we should leave out `public` if there are otherwise no naming collisions?

Reply:
In every case but as i alluded to above, i think we shouldn't support public subnets in this initial PR and try to add support for them in a followup PR due to the added complexity.

Reply:
I have removed support for Public and utility subnets.

There is still a question about the DualStack